src/cmd/cli/command/stack.go

```diff
@@ -0,0 +1,174 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	cliClient "github.com/DefangLabs/defang/src/pkg/cli/client"
+	"github.com/DefangLabs/defang/src/pkg/modes"
+	"github.com/DefangLabs/defang/src/pkg/stacks"
+	"github.com/DefangLabs/defang/src/pkg/term"
+	"github.com/spf13/cobra"
+)
+
+func makeStackCmd() *cobra.Command {
+	var stackCmd = &cobra.Command{
+		Use:   "stack",
+		Args:  cobra.NoArgs,
+		Short: "Manage Defang deployment stacks",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return nil
+		},
+	}
+	stackNewCmd := makeStackNewCmd()
+	stackCmd.AddCommand(stackNewCmd)
+	stackListCmd := makeStackListCmd()
+	stackCmd.AddCommand(stackListCmd)
+	stackRemoveCmd := makeStackRemoveCmd()
+	stackCmd.AddCommand(stackRemoveCmd)
+	return stackCmd
+}
+
+func makeStackNewCmd() *cobra.Command {
+	var stackNewCmd = &cobra.Command{
+		Use:   "new STACK_NAME",
+		Args:  cobra.ExactArgs(1),
+		Short: "Create a new Defang deployment stack",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			stackName := args[0]
+			var region, _ = cmd.Flags().GetString("region")
+
+			params := stacks.StackParameters{
+				Name:     stackName,
+				Provider: providerID, // default provider
+				Region:   region,
+				Mode:     mode,
+			}
+
+			if nonInteractive {
+				return stacks.Create(params)
+			}
+
+			if params.Provider == cliClient.ProviderAuto {
+				provider := ""
```

Comment:
I prefer `var provider string` if you want to signal that's it's not initialized. 
```suggestion
				var provider string
```